pennylane/pauli/utils.py

```diff
@@ -92,20 +93,100 @@ def is_pauli_word(observable):
     >>> is_pauli_word(4 * qml.PauliX(0) @ qml.PauliZ(0))
     True
     """
+    return False
+
+
+@is_pauli_word.register(PauliX)
+@is_pauli_word.register(PauliY)
+@is_pauli_word.register(PauliZ)
+@is_pauli_word.register(Identity)
+def _is_pw_pauli(
+    observable: Union[PauliX, PauliY, PauliZ, Identity]
+):  # pylint:disable=unused-argument
+    return True
+
+
+@is_pauli_word.register
+def _is_pw_tensor(observable: Tensor):
     pauli_word_names = ["Identity", "PauliX", "PauliY", "PauliZ"]
-    if isinstance(observable, Tensor):
-        return set(observable.name).issubset(pauli_word_names)
+    return set(observable.name).issubset(pauli_word_names)
+
+
+@is_pauli_word.register
+def _is_pw_ham(observable: Hamiltonian):
+    return False if len(observable.ops) != 1 else is_pauli_word(observable.ops[0])
+
+
+@is_pauli_word.register
+def _is_pw_prod(observable: Prod):
+    return all(is_pauli_word(op) for op in observable)
+
+
+@is_pauli_word.register
+def _is_pw_sprod(observable: SProd):
+    return is_pauli_word(observable.base)
+
+
+@singledispatch
+def pauli_word_prefactor(observable):
+    """If the operator provided is a valid Pauli word (i.e a single term which may be a tensor product
+    of pauli operators), then this function extracts the prefactor.
+
+    Args:
+        observable (~.Operator): the operator to be examined
```

Comment:
Don't we support more than just `Operator`?

Reply:
Well everything inherits from operator, so it's kind of a catch all! I can be more specific if needed.

Reply:
Right, so this function doesn't accept a Pauli word, because that always has 1 as a prefactor?